### DISCUSSION TITLE
Which way to Store Twitter oauth token in database?

### DISCUSSION THREAD

**Discussion by vaibhavarora14** [nextauthjs/next-auth#3414] [Help]
Hi,

I am using Nextjs first time for my side project.
Objective - Let users sign in to their Twitter account, store the OAuth token in our database.

What should be the approach to do that?
My options with the problem I am facing -
1) I see `callbacks` get `account -> oauth_token` information, but prisma is undefined at that time (I don't know why). Hence I am unable to store them in the database from there.
2) Twitter renders my callbackURL of the page (frontend), I will have to pass oauth_token information from there to the backend with a POST request call and save it. Although I am not able to figure, how to send this oauth_token to Frontend in the first place

Please guide me which option either of these or something better I should follow 😊

**Comment by balazsorban44** [comment]
https://next-auth.js.org/adapters/overview

https://next-auth.js.org/adapters/prisma

https://next-auth.js.org/providers/twitter

**Reply by vaibhavarora14** [reply]
I checked these again, I am not able to find answer there
